lnd.go

```diff
@@ -539,6 +567,20 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 	}
 	defer atplManager.Stop()
 
+	// If --tlsencryptkey is set, we previously generated a throwaway
+	// TLSConfig. Now we want to remove that and load the persistent
+	// TLSConfig. The wallet is unlocked at this point so we can use the
+	// real KeyRing.
+	if cfg.TLSEncryptKey {
```

Comment:
Got it, so pretty much encapsulate everything in this `if` statement (including the if statemnet)? And then expose SetCertificateAfterUnlock(KeyRing) instead of LoadPermanentCertificate?